openedx/adg/lms/applications/handlers.py

```diff
@@ -66,3 +71,40 @@ def update_prereq_courses_status(instance, **kwargs):  # pylint: disable=unused-
 
     if is_any_status_updated:
         hub.save()
+
+
+@receiver(post_save, sender=PersistentCourseGrade)
+def update_bu_prereq_courses_status_to_completed(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update bu_prereq_courses_status to COMPLETED in Application Hub
+    depending on the bu course that the learner has enrolled in
+    """
+    if instance.letter_grade and instance.passed_timestamp:
+        current_user = User.objects.get(id=instance.user_id)
+        hub = ApplicationHub.objects.filter(user=current_user).first()
+        is_any_status_updated = False
+
+        if hub.bu_prereq_courses_status == ApplicationHub.IN_PROGRESS and\
+            is_course_in_prereqs(instance.course_id, bu_prereq=True):
+            hub.bu_prereq_courses_status = ApplicationHub.COMPLETED
+            is_any_status_updated = True
+            emit_event_for_bu_courses_progress(current_user, COMPLETED)
+
+        if is_any_status_updated:
+            hub.save()
+
+            _, _, are_bu_courses_completed = (
+                get_course_cards_and_gross_details(
+                    current_user,
+                    MultilingualCourseGroup.objects.get_user_business_line_and_common_business_line_prereq_courses(
+                        current_user
+                    )
+                )
+            )
```

Comment:
This function is doing a lot of extra things. Instead of using it here and discarding the extra returns, how about creating a new function to check the if the courses are completed and using that inside the `get_course_cards_and_gross_details` function and here as well.